src/core/connect-extension/connect-extension-web.extension.ts

```diff
@@ -195,49 +197,92 @@ export const ConnectExtensionWeb = (() => {
         }
     };
 
-    const listenLastSyncForConnect = (conn: IQRCodeConn) => {
-        const connectionsRef = getRealtimeDBConnectionsRef();
+    const syncConnect = async (conn: IQRCodeConn, connectionsRef: any) => {
+        // show loading untill data is fetch and state is build
+        await LoadingModal.open();
 
-        connectionsRef.child(conn.connectionId).on('value', async (snapshot: any) => {
-            const snap = snapshot.val();
+        try {
+            // Extension the state from Firebase Storage
+            const extState = await downloadFileStorage(conn.connectionId);
 
-            if (snap?.lastSynced && snap?.authToken) {
-                // show loading untill data is fetch and state is build
-                await LoadingModal.open();
+            if (extState) {
+                const decryptedState = JSON.parse(await decrypt(extState, conn.encKey));
 
-                try {
-                    // Extension the state from Firebase Storage
-                    const extState = await downloadFileStorage(conn.connectionId);
+                // Save state
+                await storeState(decryptedState);
 
-                    if (extState) {
-                        const decryptedState = JSON.parse(await decrypt(extState, conn.encKey));
+                // Store connection
+                await storeConnection(conn);
 
-                        // Save state
-                        await storeState(decryptedState);
+                // remove listener for connectionId
+                connectionsRef.child(conn.connectionId).off('value');
+
+                // navigate to Dashboard
+                NavigationService.navigate('MainNavigation', {});
+
+                buildTransactions(decryptedState.state.wallets);
+            } else {
+                if (syncConnAttempts <= CONN_EXT_RETRY_ATTEMPTS) {
+                    // Retry
+                    syncConnect(conn, connectionsRef);
+                }
+            }
 
-                        // Store connection
-                        await storeConnection(conn);
+            // close loading modal
+            await LoadingModal.close();
 
-                        // remove listener for connectionId
-                        connectionsRef.child(conn.connectionId).off('value');
+            // Reset sync conn attempts
+            syncConnAttempts = 0;
```

Comment:
to be sure that flow will be fine you should initialise `syncConnAttempts` with `0` at the beginning of the flow, it would be more simple, but by using a global variable here if there are calls to `syncConnect` from multiple places your code will not work correctly.

To be more safe you should avoid using global vars (`syncConnAttempts`), you can achieve this by using a recursive function. e.g.
```ts
const syncConnect = async (conn: IQRCodeConn, connectionsRef: any, syncConnAttempts: number = CONN_EXT_RETRY_ATTEMPTS) => {
  try {
    if (syncConnAttempts > 0) {
      //do stuff
    } else {
      // handle error
    }
  } catch (e) {
    // something happened
    // check if retry is needed
    syncConnect(conn, connectionsRef, syncConnAttempts - 1);
  }
}